pkg/scheduler/core/equivalence/eqivalence.go

```diff
@@ -287,6 +330,7 @@ func (n *NodeCache) invalidatePreds(predicateKeys sets.String) {
 	defer n.mu.Unlock()
 	for predicateKey := range predicateKeys {
 		delete(n.cache, predicateKey)
+		n.generations[predicateKey]++
```

Comment:
Also, at line 153, all predicates of a node are invalidated. We currently just delete the node from the cache. With this PR, we can no longer delete the node. We must keep the node, increment generation number of all the predicates and set the `predicateMap` to nil (or an empty map).